src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts

```diff
@@ -22,99 +31,139 @@ export class TransferImitationMapper {
-        !isErc20Transfer(item.transaction.txInfo.transferInfo) ||
-        !isErc20Transfer(prevItem.transaction.txInfo.transferInfo)
+        !isTransferTransactionInfo(txInfo) ||
+        !isErc20Transfer(txInfo.transferInfo)
       ) {
-        return item;
+        mappedTransactions.unshift(item);
+        continue;
       }
 
-      // ...that are outgoing
-      const isOutgoing =
-        item.transaction.txInfo.direction === TransferDirection.Outgoing;
-      const isPrevOutgoing =
-        prevItem.transaction.txInfo.direction === TransferDirection.Outgoing;
-      if (!isOutgoing || !isPrevOutgoing) {
-        return item;
+      /**
+       * Transactions to compare for imitation against, limited by a lookup distance.
+       *
+       * Concatenation takes preference of already mapped transactions over their
+       * original in order to prevent comparison against duplicates.
+       * Its length is {@link transactions} + 1 as {@link previousTransaction}
+       * is appended to compare {@link transactions.at(-1)} against.
+       */
+      const prevItems = mappedTransactions
+        .concat(args.transactions.slice(i - 1), args.previousTransaction ?? [])
+        // Only compare so far back
+        .slice(0, this.lookupDistance);
+
+      if (prevItems.length === 0) {
+        mappedTransactions.unshift(item);
+        continue;
       }
 
-      // Imitation transfers are of the same value...
-      const isSameValue =
-        item.transaction.txInfo.transferInfo.value ===
-        prevItem.transaction.txInfo.transferInfo.value;
-      if (!isSameValue) {
-        return item;
+      // Imitation transfers often employ differing decimals to prevent direct
+      // comparison of values. Here we normalize the value
+      const formattedValue = this.formatValue(
+        txInfo.transferInfo.value,
+        txInfo.transferInfo.decimals,
+      );
+      // Either sender or recipient according to "direction" of transaction
+      const refAddress = this.getReferenceAddress(txInfo);
+
+      const isImitation = prevItems.some((prevItem) => {
+        const prevTxInfo = prevItem.transaction.txInfo;
+        if (
+          !isTransferTransactionInfo(prevTxInfo) ||
+          !isErc20Transfer(prevTxInfo.transferInfo) ||
+          // Do not compare against previously identified imitations
+          prevTxInfo.transferInfo.imitation
+        ) {
+          return false;
+        }
+
+        const prevFormattedValue = this.formatValue(
+          prevTxInfo.transferInfo.value,
+          prevTxInfo.transferInfo.decimals,
+        );
+
+        // Imitation transfers match in value
+        if (formattedValue !== prevFormattedValue) {
+          return false;
+        }
+
+        // Imitation transfers match in vanity of address
+        const prevRefAddress = this.getReferenceAddress(prevTxInfo);
+        return this.isImitatorAddress(refAddress, prevRefAddress);
+      });
+
+      txInfo.transferInfo.imitation = isImitation;
```

Comment:
Nitpick: if the `for` loop `continue` before reaching this point, the value of `txInfo.transferInfo.imitation` is `null`. Do you see value in setting it to `false` in those situations? Maybe the API of this mapper is simpler if we enforce a boolean type in the field?

Reply:
In 5a945cd1c5b703430c253115bc220e94438dfc82 I changed the default value of `Erc20Transfer['imitation']` to `false` in the and explicitly set it here as requested.